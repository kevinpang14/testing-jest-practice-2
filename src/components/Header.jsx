import React from "react";

export default function Header() {
  return <h1>Welcome to my Apps</h1>;
}
