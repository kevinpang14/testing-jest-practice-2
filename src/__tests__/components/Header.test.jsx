import { render } from "@testing-library/react";

import Header from "../../components/Header";

describe("Header Component", () => {
  test("check h1 tag", () => {
    render(<Header />);
    expect(document.querySelector("h1")).toBeInTheDocument();
  });

  test("match snapshot", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
