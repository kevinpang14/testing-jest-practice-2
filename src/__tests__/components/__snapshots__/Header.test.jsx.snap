// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component match snapshot 1`] = `
<div>
  <h1>
    Welcome to my Apps
  </h1>
</div>
`;
